scripts/prep/rtofs/exevs_rtofs_prep.sh

```diff
@@ -138,86 +141,86 @@ min_size=2404
+		    	echo "Warning: No OSI-SAF ${ftype} data was available for valid date $VDATE." > mailmsg
+		    	echo "Missing file is $input_osisaf_file" >> mailmsg
+		    	cat mailmsg | mail -s "$subject" $MAILTO
+	    	fi
+    	fi
+    	if [ ! -s $output_osisaf_file ]; then
+        	if [ -s $input_osisaf_file ]; then
+            		cdo remapbil,$osi_saf_grid_file $input_osisaf_file $tmp_osisaf_file
+            		export err=$?; err_chk
+            		if [ $SENDCOM = "YES" ]; then
+                		if [ -s $tmp_osisaf_file ]; then
+		    			cp -v $tmp_osisaf_file $output_osisaf_file
+				fi
+            		fi
+        	else
+			echo "WARNING: No OSI-SAF ${ftype} data was available for valid date $VDATE."
+            		if [ $SENDMAIL = YES ] ; then
+                		export subject="OSI-SAF Data Missing for EVS RTOFS"
+                		echo "Warning: No OSI-SAF ${ftype} data was available for valid date $VDATE." > mailmsg
+                		echo "Missing file is $input_osisaf_file" >> mailmsg
+                		cat mailmsg | mail -s "$subject" $MAILTO
+            		fi
+        	fi
+    	fi
 done
 # convert NDBC *.txt files into a netcdf file using ASCII2NC
 export RUN=ndbc
 if [ ! -d $COMOUTprep/rtofs.$VDATE/$RUN ]; then
-    mkdir -p $COMOUTprep/rtofs.$VDATE/$RUN
+	mkdir -p $COMOUTprep/rtofs.$VDATE/$RUN
 fi
 if [ ! -d $COMOUTprep/rtofs.$VDATE/$RUN/buoy ]; then
-    mkdir -p $COMOUTprep/rtofs.$VDATE/$RUN/buoy
+    	mkdir -p $COMOUTprep/rtofs.$VDATE/$RUN/buoy
 fi
 mkdir -p $DATA/rtofs.$VDATE/$RUN
 mkdir -p $DATA/rtofs.$VDATE/$RUN/buoy
 export MET_NDBC_STATIONS=${FIXevs}/ndbc_stations/ndbc_stations.xml
 ndbc_txt_ncount=$(find $DCOMROOT/$VDATE/validation_data/marine/buoy -type f -name "*.txt" |wc -l)
 if [ $ndbc_txt_ncount -gt 0 ]; then
-    python $USHevs/${COMPONENT}/${COMPONENT}_${STEP}_trim_ndbc_files.py
-    export err=$?; err_chk
-    tmp_ndbc_file=$DATA/rtofs.$VDATE/$RUN/ndbc.${VDATE}.nc
-    output_ndbc_file=$COMOUTprep/rtofs.$VDATE/$RUN/ndbc.${VDATE}.nc
-    if [ ! -s $output_ndbc_file ]; then
-        run_metplus.py -c $PARMevs/metplus_config/machine.conf \
-        -c $CONFIGevs/$STEP/$COMPONENT/grid2obs/ASCII2NC_obsNDBC.conf
-        export err=$?; err_chk
-         if [ $SENDCOM = YES ]; then
-             if [ -s $tmp_ndbc_file ] ; then
-		 cp -v $tmp_ndbc_file $output_ndbc_file
-	     fi
-         fi
-    fi
+	python $USHevs/${COMPONENT}/${COMPONENT}_${STEP}_trim_ndbc_files.py
+    	export err=$?; err_chk
+    	tmp_ndbc_file=$DATA/rtofs.$VDATE/$RUN/ndbc.${VDATE}.nc
+    	output_ndbc_file=$COMOUTprep/rtofs.$VDATE/$RUN/ndbc.${VDATE}.nc
+    	if [ ! -s $output_ndbc_file ]; then
+        	run_metplus.py -c $PARMevs/metplus_config/machine.conf \
+        	-c $CONFIGevs/$STEP/$COMPONENT/grid2obs/ASCII2NC_obsNDBC.conf
+        	export err=$?; err_chk
+         	if [ $SENDCOM = YES ]; then
+             		if [ -s $tmp_ndbc_file ] ; then
+				cp -v $tmp_ndbc_file $output_ndbc_file
+			fi
+         	fi
+   	 fi
 else
```

Comment:
@SamiraArdani-NOAA Another question.  We have all sorts of spacing changes such as these.  Is there a reason for so many spacing changes like this?

Reply:
@PerryShafran-NOAA : No reason. It is just for my convenience since we have so many if-statements and loops in the code, I set those tab spacing to make sure I am adding the changes at the right place.